### PR TITLE
add tests for routing

### LINKED ISSUE #28: tubes Router is broken
Besides the broken docstring for the Router as reported in #21 the Router also generally appears not to work; or am I missing something simple here?

see my attempt at creating unit tests for Router:
https://github.com/david415/tubes/tree/add_routing_tests.0
https://github.com/david415/tubes/blob/add_routing_tests.0/tubes/test/test_routing.py

When the Router's newRoute() method is called this results in:

```
$ trial tubes.test.test_routing
tubes.test.test_routing
  TestBasicRouter
    test_basic_router ...                                               [ERROR]

===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/user/tubes/tubes/test/test_routing.py", line 64, in test_basic_router
    evenOddTube.addRoutes()
  File "/home/user/tubes/tubes/test/test_routing.py", line 41, in addRoutes
    self.evenRoute = self.newRoute()
  File "/home/user/tubes/tubes/routing.py", line 179, in newRoute
    fount = self._out.newFount().flowTo(received)
  File "/home/user/tubes/tubes/fan.py", line 253, in flowTo
    return beginFlowingTo(self, drain)
  File "/home/user/tubes/tubes/kit.py", line 104, in beginFlowingTo
    return drain.flowingFrom(fount)
exceptions.AttributeError: '_Tubule' object has no attribute 'flowingFrom'

tubes.test.test_routing.TestBasicRouter.test_basic_router
-------------------------------------------------------------------------------
Ran 1 tests in 0.018s

FAILED (errors=1)
```

### PR DESCRIPTION
**Pull request by glyph**
Fixes #28

[Review on Reviewable](https://reviewable.io/reviews/twisted/tubes/32)

### CODE REVIEW

**Comment by glyph** [comment]
@david415 - hopefully router.py won't be a broken mess after this...

**Comment by glyph** [comment]
Reviewed 5 of 6 files at r1.
Review status: 5 of 6 files reviewed at latest revision, 1 unresolved discussion.

---

_[tubes/routing.py, line 114 \[r1\]](https://reviewable.io:443/reviews/twisted/tubes/32#-K3i0sEkLoduOODn74_b:-K3i0sEldf5gnYYdbwAI:2020182402) ([raw file](https://github.com/twisted/tubes/blob/f2a9ad7bc70db219e25989953dacd8ab712f433a/tubes/routing.py#L114)):_
Needs an __ne__

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/twisted/tubes/32)_

File: tubes/routing.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# -*- test-case-name: tubes.test.test_fan -*-
+# -*- test-case-name: tubes.test.test_routing -*-
 # Copyright (c) Twisted Matrix Laboratories.
 # See LICENSE for details.
```
Comment (david415) [inline tubes/routing.py]: This doc string isn't very accurate; we should demonstrate that the message destination must be added to the message before being sent to the router. In `test_twoRoutes` this is made clear with routeDrain constructed like this :

```
routeDrain = series(chooser, router.drain)
```

**Comment by glyph** [comment]
*[tubes/routing.py, line 4 \[r1\]](https://reviewable.io:443/reviews/twisted/tubes/32#-K3gWxSrSFLp176A0YkX-r1-4:-K3wspSB0UkMDlRNrCud:1502677532) ([raw file](https://github.com/twisted/tubes/blob/f2a9ad7bc70db219e25989953dacd8ab712f433a/tubes/routing.py#L4)):*
Thanks for catching this; I'll fix it.  I think in the future we should probably filter stuff like this out into separate tickets though, so we can land minor improvements without requiring re-review.  But we are still in a phase with tubes where self-review is acceptable so I'll just self-review it for now ;)

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/twisted/tubes/32)*

**Comment by glyph** [comment]
Reviewed 1 of 6 files at r1.
Review status: 4 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[tubes/routing.py, line 4 \[r1\]](https://reviewable.io:443/reviews/twisted/tubes/32#-K3gWxSrSFLp176A0YkX-r1-4:-K3xOQlUwQM-1nRHHAj6:-499146009) ([raw file](https://github.com/twisted/tubes/blob/f2a9ad7bc70db219e25989953dacd8ab712f433a/tubes/routing.py#L4)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/twisted/tubes/32)*

**Comment by glyph** [comment]
Reviewed 2 of 2 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/twisted/tubes/32)*

**Comment by glyph** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/twisted/tubes/32)*

**Comment by glyph** [comment]
*[tubes/routing.py, line 4 \[r1\]](https://reviewable.io:443/reviews/twisted/tubes/32#-K3gWxSrSFLp176A0YkX-r1-4:-K3xObh9nYm8vFiQxLQ8:-499146009) ([raw file](https://github.com/twisted/tubes/blob/f2a9ad7bc70db219e25989953dacd8ab712f433a/tubes/routing.py#L4)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/twisted/tubes/32)*

**Comment by glyph** [comment]
OK, so this is kind of a dealbreaker on @reviewable; assent from the *original reviewer* is required in order to make the status check go green; nobody else can resolve the comment.  Given how rarely a single Twisted reviewer can carry a review to completion, I think we'll have to remove it as a required status check.

**Pull request merged**